core/ir/x86/decode.c

```diff
@@ -2410,6 +2436,22 @@ decode_get_tuple_type_input_size(const instr_info_t *info, decode_info_t *di)
         di->input_size = OPSZ_NA;
 }
 
+/* TODO i#6238: Not all opcodes have been reviewed.
+ * In case an opcode has not been reviewed,
+ * the default category assigned to it is DR_INSTR_CATEGORY_OTHER.
+ */
+static inline void
+decode_category(instr_t *instr)
+{
+    if (instr) {
+        if (op_instr[instr->opcode])
+            instr_set_category(instr, op_instr[instr->opcode]->category);
+        else
+            /* nonvalid opcode */
+            instr_set_category(instr, DR_INSTR_CATEGORY_UNCATEGORIZED);
+    }
+}
+
```

Comment:
I'm seeing a bunch of entries that I think have the wrong category.  Here are some:

>+    {OP_push, 0x060000, DR_INSTR_CATEGORY_OTHER, "push", xsp, i_xSPo1, es, xsp, xx, i64, x, tfb[0x0e]},

This should be a _STORE.  (Some OP_push variants can also be a _LOAD but we let the C code add that I guess since I think it's another type E.)

>+    {OP_mov_imm, 0xb00000, DR_INSTR_CATEGORY_OTHER, "mov", al_x, xx, Ib, xx, xx, no, x, tfb[0xb1]},

Shouldn't this be _MOVE?

>+    {OP_adc,  0x100000, DR_INSTR_CATEGORY_OTHER, "adc",  Eb, xx, Gb, Eb, xx, mrm, (fW6|fRC), tex[1][2]},

This is an addition with carry so it should be _INT|_MATH.

If you don't want to go update all of these, which is a bunch of work, maybe for this initial PR just replace all the _OTHER with _UNCATEGORIZED?  Then a later PR can fix up the uncats and it's clear what's missing.

Reply:
ok. replaced to _UNCATEGORIZED